pysgrid/read_netcdf.py

```diff
@@ -150,11 +232,248 @@ def sgrid_compliant_file(self):
         
         """
         grid_vars = self.find_grid_topology_vars()
-        if len(grid_vars) > 0:
+        if grid_vars is not None:
             sgrid_compliant = True
         else:
             sgrid_compliant = False
         return sgrid_compliant
+    
+    
+class SGridND(object):
+    
+    topology_dim = None
+    
+    def __init__(self, sgrid, nc_dataset, topology_variable):
+        self._sgrid = sgrid
+        self.nc_dataset = nc_dataset
+        self.ncd = NetCDFDataset(self.nc_dataset)
+        self.topology_variable = topology_variable  # the netCDF variable with a cf_role of 'grid_topology'
+        self.topology_var = self.nc_dataset.variables[self.topology_variable]
+        self.pp = ParsePadding(self.topology_variable)
+        
+    @property
+    def sgrid(self):
+        return self._sgrid
+    
+    def set_dimensions(self):
+        ds_dims = self.nc_dataset.dimensions
+        grid_dims = [(ds_dim, len(ds_dims[ds_dim])) for ds_dim in ds_dims]
+        self._sgrid.dimensions = grid_dims
+        
+    def set_topology_dimension(self):
+        self._sgrid.topology_dimension = self.topology_dim
+    
+    def set_edge1_dimensions(self):
+        try:
+            edge1_dim = self.topology_var.edge1_dimensions
+            edge1_dim_padding = self.pp.parse_padding(edge1_dim)
+            self._sgrid.edge_1_dimension = edge1_dim
+            self._sgrid.edge_1_padding = edge1_dim_padding
+        except AttributeError:
+            pass
```

Comment:
I guess I failed in my example.  My concern is that you are expecting an `AttributeError` from one (or both) of these:

```python
edge1_dim = self.topology_var.edge1_dimensions
edge1_dim_padding = self.pp.parse_padding(edge1_dim)
```

and I am unsure if silently passing it is the right way to go.  Anyways, the PR is good to merge.  That is just something we can discuss later as I learn more about the code.

The danger of silently passing Exceptions is that, for example, `parse_padding()` is a method that might not may not make use of attribute right now.  The code is expecting that exception only from  `self.topology_var.edge1_dimensions`.  However, if in the future, `parse_padding()` uses an attribute that might fails we'll never know what happened.